lib/notificationhub.go

```diff
@@ -0,0 +1,295 @@
+package lib
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/daresaydigital/azure-notificationhubs-go/lib/utils"
+	"gopkg.in/xmlpath.v2"
+)
+
+// NotificationHub is a client for sending messages through Azure Notification Hubs
+type NotificationHub struct {
+	SasKeyValue string
+	SasKeyName  string
+	HubURL      *url.URL
+
+	client                  utils.HTTPClient
+	expirationTimeGenerator utils.ExpirationTimeGenerator
+	regIDPath               *xmlpath.Path
+	eTagPath                *xmlpath.Path
+	expTmPath               *xmlpath.Path
+}
+
+// NewNotificationHub initializes and retubrns NotificationHub pointer
+func NewNotificationHub(connectionString, hubPath string) *NotificationHub {
+	var (
```

Comment:
Looks like some of these can be declared as consts instead?

Reply:
👍